src/sockaddr.rs

```diff
@@ -30,20 +30,38 @@ use std::net;
 
 use crate::bindings;
 
-pub fn to_sockaddr(addr: net::Ipv4Addr) -> bindings::sockaddr {
-    let mut sockaddr_in: nix::libc::sockaddr_in = unsafe { std::mem::zeroed() };
+pub use bindings::sockaddr as sockaddr;
 
-    sockaddr_in.sin_family = nix::libc::AF_INET as u16;
-    sockaddr_in.sin_addr = nix::libc::in_addr {
-        s_addr: u32::from_le_bytes(addr.octets()),
-    };
-    sockaddr_in.sin_port = 0;
-
-    unsafe { std::mem::transmute(sockaddr_in) }
+impl From<net::Ipv4Addr> for bindings::sockaddr {
+    fn from(addr: net::Ipv4Addr) -> Self {
+        let mut sockaddr_in: nix::libc::sockaddr_in = unsafe { std::mem::zeroed() };
+    
+        sockaddr_in.sin_family = nix::libc::AF_INET as u16;
+        sockaddr_in.sin_addr = nix::libc::in_addr {
+            s_addr: u32::from_le_bytes(addr.octets()),
+        };
+        sockaddr_in.sin_port = 0;
+    
+        unsafe { std::mem::transmute(sockaddr_in) }
+    }
 }
+impl From<[u8; 6]> for bindings::sockaddr {
+    fn from(addr: [u8; 6]) -> Self {
+        let mut addr = addr.to_vec();
+        addr.append(&mut [0x00; 8].to_vec());
+        
+        bindings::sockaddr { sa_family: nix::libc::ARPHRD_ETHER, sa_data: addr.iter().map(|x| *x as i8).collect::<Vec<i8>>().try_into().unwrap() }
```

Comment:
I'm not sure we should directly work with `sockaddr`. The reason that we have to pad the address with zeros in the first place is because `sockaddr` is a generic socket address. As in `From<net::Ipv4Addr>`, I think we should use `sockadd_ll` here that has the following definition:

```c
struct sockaddr_ll
  {
    unsigned short int sll_family;
    unsigned short int sll_protocol;
    int sll_ifindex;
    unsigned short int sll_hatype;
    unsigned char sll_pkttype;
    unsigned char sll_halen;
    unsigned char sll_addr[8];
  };
```

It has `sll_addr[8]` which is exactly what we want. Then we can transmute it to `sockaddr` when returning.

I'm not sure about the correct way to initialize this struct so some research might be necessary :)